In [7]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots
import plotly.io as pio
import plotly.colors as colors
pio.templates.default = "plotly_white"


In [8]:
try:
    df = pd.read_csv('/content/Superstore.csv', encoding='latin1')
except UnicodeDecodeError:
    df = pd.read_csv('/content/Superstore.csv', encoding='ISO-8859-1')

df.head()

Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1       42420   South  FUR-CH-10000454        Furniture       Chairs   
2       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3       33311   South  FUR-TA-10000577        Furniture       Tables   
4       33311   South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                     Eldon Fold 'N Roll Cart System   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164  

[5 rows x 21 columns]

In [9]:
df.shape

(9994, 21)

In [10]:
df.describe()

Row ID   Postal Code         Sales     Quantity     Discount  \
count  9994.000000   9994.000000   9994.000000  9994.000000  9994.000000   
mean   4997.500000  55190.379428    229.858001     3.789574     0.156203   
std    2885.163629  32063.693350    623.245101     2.225110     0.206452   
min       1.000000   1040.000000      0.444000     1.000000     0.000000   
25%    2499.250000  23223.000000     17.280000     2.000000     0.000000   
50%    4997.500000  56430.500000     54.490000     3.000000     0.200000   
75%    7495.750000  90008.000000    209.940000     5.000000     0.200000   
max    9994.000000  99301.000000  22638.480000    14.000000     0.800000   

            Profit  
count  9994.000000  
mean     28.656896  
std     234.260108  
min   -6599.978000  
25%       1.728750  
50%       8.666500  
75%      29.364000  
max    8399.976000

In [11]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [12]:
df.duplicated().sum()

0

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [14]:
def filter_columns_by_type(df):
  numerical_columns = df.select_dtypes(include=['number']).columns.tolist()
  object_columns = df.select_dtypes(include=['object']).columns.tolist()
  return numerical_columns, object_columns

numerical_cols, object_cols = filter_columns_by_type(df)
print("Numerical Columns:", numerical_cols)
print("Object Columns:", object_cols)

Numerical Columns: ['Row ID', 'Postal Code', 'Sales', 'Quantity', 'Discount', 'Profit']
Object Columns: ['Order ID', 'Order Date', 'Ship Date', 'Ship Mode', 'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State', 'Region', 'Product ID', 'Category', 'Sub-Category', 'Product Name']


In [15]:
# Converting datetime
df['Order Date'] = pd.to_datetime(df['Order Date'])
df['Ship Date'] = pd.to_datetime(df['Ship Date'])

In [16]:
#New Column dt.year, dt.month, dt.daysofweek
df['Order Year'] = df['Order Date'].dt.year
df['Order Month'] = df['Order Date'].dt.month
df['Order Day of Week'] = df['Order Date'].dt.dayofweek

# Month Sales Analysis

In [17]:
sales_by_month = df.groupby('Order Month')['Sales'].sum().reset_index()

In [18]:
sales_by_month

Order Month        Sales
0             1   94924.8356
1             2   59751.2514
2             3  205005.4888
3             4  137762.1286
4             5  155028.8117
5             6  152718.6793
6             7  147238.0970
7             8  159044.0630
8             9  307649.9457
9            10  200322.9847
10           11  352461.0710
11           12  325293.5035

In [19]:
#Visulization
fig = px.line(sales_by_month, x='Order Month', y='Sales', title='Monthly Sales Trend')
fig.show()

# Sales by Category

In [20]:
sales_by_category = df.groupby('Category')['Sales'].sum().reset_index().sort_values(by='Sales', ascending=False)
sales_by_category

Category        Sales
2       Technology  836154.0330
0        Furniture  741999.7953
1  Office Supplies  719047.0320

In [21]:
# Visualization pie
fig = px.pie(sales_by_category, names='Category', values='Sales', title='Sales by Category' )
fig.show()


# Sales Analysis by Sub- Category

In [22]:
sales_sub_category = df.groupby('Sub-Category')['Sales'].sum().reset_index().sort_values(by='Sales', ascending=False)
sales_sub_category

Sub-Category        Sales
13       Phones  330007.0540
5        Chairs  328449.1030
14      Storage  223843.6080
16       Tables  206965.5320
3       Binders  203412.7330
11     Machines  189238.6310
0   Accessories  167380.3180
6       Copiers  149528.0300
4     Bookcases  114879.9963
1    Appliances  107532.1610
9   Furnishings   91705.1640
12        Paper   78479.2060
15     Supplies   46673.5380
2           Art   27118.7920
7     Envelopes   16476.4020
10       Labels   12486.3120
8     Fasteners    3024.2800

In [23]:
#Visualization
fig = px.bar(sales_sub_category, x='Sub-Category', y='Sales', title='Sales by Sub-Category')
fig.show()

# Monthly Profit Analysis

In [24]:
profit_by_month = df.groupby('Order Month')['Profit'].sum().reset_index()
profit_by_month

Order Month      Profit
0             1   9134.4461
1             2  10294.6107
2             3  28594.6872
3             4  11587.4363
4             5  22411.3078
5             6  21285.7954
6             7  13832.6648
7             8  21776.9384
8             9  36857.4753
9            10  31784.0413
10           11  35468.4265
11           12  43369.1919

In [25]:
#Visualization
fig = px.line(profit_by_month, x='Order Month', y='Profit', title='Monthly Profit Trend')
fig.show()

In [26]:
profit_by_category = df.groupby('Category')['Profit'].sum().reset_index().sort_values(by='Profit', ascending=False)
profit_by_category

Category       Profit
2       Technology  145454.9481
1  Office Supplies  122490.8008
0        Furniture   18451.2728

In [27]:
# Visualization
fig = px.pie(profit_by_category, names='Category', values='Profit', title='Profit by Category')
fig.show()

In [28]:
profit_by_sub_category = df.groupby('Sub-Category')['Profit'].sum().reset_index().sort_values(by='Profit', ascending=False)
profit_by_sub_category

Sub-Category      Profit
6       Copiers  55617.8249
13       Phones  44515.7306
0   Accessories  41936.6357
12        Paper  34053.5693
3       Binders  30221.7633
5        Chairs  26590.1663
14      Storage  21278.8264
1    Appliances  18138.0054
9   Furnishings  13059.1436
7     Envelopes   6964.1767
2           Art   6527.7870
10       Labels   5546.2540
11     Machines   3384.7569
8     Fasteners    949.5182
15     Supplies  -1189.0995
4     Bookcases  -3472.5560
16       Tables -17725.4811

In [29]:
# Visualization
fig = px.bar(profit_by_sub_category, x='Sub-Category', y='Profit', title='Profit by Sub-Category')
fig.show()

In [31]:
sales_profit_by_segment = df.groupby('Segment')[['Sales', 'Profit']].sum().reset_index()
sales_profit_by_segment


Segment         Sales       Profit
0     Consumer  1.161401e+06  134119.2092
1    Corporate  7.061464e+05   91979.1340
2  Home Office  4.296531e+05   60298.6785

In [33]:
import plotly.graph_objects as go

# Create figure
fig = go.Figure()

# Add Sales bar trace
fig.add_trace(go.Bar(x=sales_profit_by_segment['Segment'],
                     y=sales_profit_by_segment['Sales'],
                     name='Sales'))

# Add Profit bar trace
fig.add_trace(go.Bar(x=sales_profit_by_segment['Segment'],
                     y=sales_profit_by_segment['Profit'],
                     name='Profit'))

# Update layout
fig.update_layout(barmode='group', title='Sales and Profit by Segment')

# Show the figure
fig.show()


# Analysis Sales to Profit Ratio

In [34]:
sales_profit_by_segment = df.groupby('Segment')[['Sales', 'Profit']].sum().reset_index()
sales_profit_by_segment['Sales_Profit_Ratio'] = sales_profit_by_segment['Sales'] / sales_profit_by_segment['Profit']
print(sales_profit_by_segment[['Segment', 'Sales_Profit_Ratio']])

       Segment  Sales_Profit_Ratio
0     Consumer            8.659471
1    Corporate            7.677245
2  Home Office            7.125416


# Conclusion

**1.Monthly Sales Overview**
The store experienced its highest sales in December ($43.4K), September ($36.9K), and November ($35.5K), while the lowest sales were recorded in January ($9.1K) and February ($10.3K), highlighting key seasonal trends.

**2.Month with Highest Sales**
December ($43.4K), September ($36.9K), and November ($35.5K) recorded the highest sales.

**3.Month with Lowest Sales**
January ($9.1K) and February ($10.3K) had the lowest sales, indicating seasonal fluctuations.

**4.Profit by Category**
Technology ($145.5K) and Office Supplies ($122.5K) drive the highest profits, while Furniture ($18.5K) shows a significant lag, suggesting room for margin improvement.

**5.Profit by Sub-Category**
Copiers ($55.6K), Phones ($44.5K), and Accessories ($41.9K) are the top profit generating sub-categories. Tables (-$17.7K), Bookcases (-$3.5K), and Supplies (-$1.2K) reported losses, indicating areas needing cost control.

**6.Sales and Profit Summary**
The Consumer segment leads in both sales ($1.16M) and profit ($134K), followed by Corporate ($706K sales, $92K profit). The Home Office segment has the lowest figures ($430K sales, $60K profit), suggesting room for improvement.

**7.Sales to Profit Ratio**
Consumer has the highest efficiency (8.66), followed by Corporate (7.68). Home Office has the lowest ratio (7.13), suggesting potential for cost optimization.

